CRAN Solaris error 3/19

Email...

> Please see the problems shown on
> <https://cran.r-project.org/web/checks/check_results_errum.html>.

`pow()` requires `doubles` not `int`. This was taken care of in ecdmcore... Just not propagated out. 

Log results... 

> errum_cpp_routine.cpp:8:28: error: call of overloaded ‘pow(int, unsigned int)’ is ambiguous
         vv(k) = pow(2,K-k-1);

> errum_cpp_routine.cpp: In function ‘arma::vec inv_bijectionvector(unsigned int, double)’:
errum_cpp_routine.cpp:17:36: error: call of overloaded ‘pow(int, unsigned int)’ is ambiguous
         double twopow = pow(2,K-k-1);

> errum_cpp_routine.cpp: In function ‘arma::mat random_Q(unsigned int, unsigned int)’:
errum_cpp_routine.cpp:73:34: error: call of overloaded ‘pow(int, unsigned int&)’ is ambiguous
     unsigned int nClass = pow(2,K);
                                  ^

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -1,6 +1,6 @@
 Package: errum
 Title: Exploratory Reduced Reparameterized Unified Model Estimation
-Version: 0.0.2
+Version: 0.0.3
 Authors@R: c(
     person("James Joseph", "Balamuta", 
            email = "balamut2@illinois.edu", 

--- a/DESCRIPTION
+++ b/DESCRIPTION
@@ -27,7 +27,7 @@ LinkingTo: Rcpp, RcppArmadillo (>= 0.9.200)
 Suggests: 
     simcdm
 LazyData: true
-RoxygenNote: 7.0.2
+RoxygenNote: 7.1.0
 Roxygen: list(markdown = TRUE)
 Encoding: UTF-8
 Language: en-US

--- a/NEWS.md
+++ b/NEWS.md
@@ -1,4 +1,10 @@
-# errum 0.0.1
+# errum 0.0.3
+
+## Bug fix
+
+- Fix ambiguous overloaded of `pow` on Solaris.
+
+# errum 0.0.2
 
 ## Features
 

--- a/cran-comments.md
+++ b/cran-comments.md
@@ -6,7 +6,9 @@
 
 ## R CMD check results
 
-0 errors | 0 warnings | 0 notes
+0 errors | 0 warnings | 1 note
+
+> Days since last update: 0
 
 - This release addresses an issue uncovered with `pow()` on Solaris causing 
   the package to fail to build.

--- a/cran-comments.md
+++ b/cran-comments.md
@@ -6,18 +6,7 @@
 
 ## R CMD check results
 
-0 errors | 0 warnings | 1 note
+0 errors | 0 warnings | 0 notes
 
-New submission
-
-Possibly mis-spelled words in DESCRIPTION:
-  Culpepper (19:56)
-  ErRUM (19:36)
-  
-- This is a new release.
-- Based on feedback from a previous submission:
-  - We've shorten the title from 75 characters to 60 characters.
-  - One of the reported misspelling is last name of one of the paper's authors and
-    the second misspelling is the acronym name for the method.
-  - We've downsampled an estimation of the example as much as possible. The 
-    example should CRAN's 5 second limitation.
+- This release addresses an issue uncovered with `pow()` on Solaris causing 
+  the package to fail to build.

--- a/src/errum_cpp_routine.cpp
+++ b/src/errum_cpp_routine.cpp
@@ -6,7 +6,7 @@ arma::vec bijectionvector(unsigned int K)
 {
     arma::vec vv(K);
     for (unsigned int k = 0; k < K; k++) {
-        vv(k) = pow(2, K - k - 1);
+        vv(k) = std::pow(2.0, static_cast<double>(K - k) - 1.0);
     }
     return vv;
 }
@@ -16,7 +16,7 @@ arma::vec inv_bijectionvector(unsigned int K, double CL)
 {
     arma::vec alpha(K);
     for (unsigned int k = 0; k < K; k++) {
-        double twopow = pow(2, K - k - 1);
+        double twopow = std::pow(2.0, static_cast<double>(K - k) - 1.0);
         alpha(k) = (twopow <= CL);
         CL = CL - twopow * alpha(k);
     }
@@ -87,7 +87,9 @@ arma::vec rDirichlet(const arma::vec &deltas)
 // [[Rcpp::export]]
 arma::mat random_Q(unsigned int J, unsigned int K)
 {
-    unsigned int nClass = pow(2, K);
+    unsigned int nClass =
+        static_cast<unsigned int>(std::pow(2.0, static_cast<double>(K)));
+
     arma::vec vv = bijectionvector(K);
     arma::vec Q_biject(J);
     Q_biject(arma::span(0, K - 1)) = vv;
@@ -490,7 +492,8 @@ Rcpp::List rRUM_mvnQ_Gibbs(const arma::mat &Y, unsigned int K,
     // Parameter initialization
     unsigned int N = Y.n_rows;
     unsigned int J = Y.n_cols;
-    unsigned int nClass = pow(2, K);
+    unsigned int nClass =
+        static_cast<unsigned int>(std::pow(2.0, static_cast<double>(K)));
 
     // Chain
     unsigned int chain_p_burn = chain_length + burnin;

